### PR TITLE
#159691579 A user can bookmark articles

### PR DESCRIPTION
**Pull request by dcDalin**
#### What does this PR do?
Allow users to bookmark articles.

#### Description of tasks to be completed
Users can bookmark articles for later reading.

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans.git
```
```sh
$ git checkout ft-bookmark-articles-159691579
```
```sh
$ cd ah-titans/authors
```

> Add **.env** file and copy **example.env** content.
> Add your local database configuration settings to your **.env** file

```sh
$ source .env
```
```sh
$ python3.6 -m venv venv
```
```sh
$ source venv/bin/activate
```
```sh
$ pip install -r requirements.txt
```
```sh
$ python manage.py makemigrations
```
```sh
$ python manage.py runserver
```

#### Any background context you want to provide?
No

#### What are the relevant Pivotal Tracker Stories?
[#159691579](https://www.pivotaltracker.com/story/show/159691579)

#### Screenshots
1_success_bookmark
2_failed_to_bookmark
3_remove_from_bookmarks
4_remove_non_existent

bookmark ui

### CODE REVIEW

**Comment by migot01** [comment]
Please add UI templates for your feature

File: authors/apps/articles/tests/test_bookmarks.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        self.create_article(token, self.article_data)
+        response = self.client.delete(
+            '/api/articles/wrong-slug/bookmark/', 
+            HTTP_AUTHORIZATION='Token ' + token,
+            format='json'
+        )
+        self.assertIn("An article with this slug does not exist",
+                      response.content.decode())
+        self.assertEquals(response.status_code,
+                          status.HTTP_404_NOT_FOUND)
```
Comment (migot01) [inline authors/apps/articles/tests/test_bookmarks.py]: Add an end of line on test_bookmarks.py line 146

**Review by janetnim — COMMENTED** [review]
add end of line in renderers.py

**Review by janetnim — APPROVED** [review]

**Review by ronaldndirangu — APPROVED** [review]
Everything looks good. I think it is ready to merge.

File: authors/apps/articles/serializers.py [hunk 2]
```diff
@@ -74,8 +73,9 @@ class ArticleSerializer(serializers.ModelSerializer):
     tagList = TagRelatedField(many=True, required=False, source='tags')
     favorited = serializers.SerializerMethodField(method_name="is_favorited")
     favoriteCount = serializers.SerializerMethodField(
-                            method_name='get_favorite_count'
-                            )
+        method_name='get_favorite_count'
+    )
+    bookmarked = serializers.SerializerMethodField(method_name="is_bookmarked")
```
Comment (ridgekimani) [inline authors/apps/articles/serializers.py:78]: Does this mean you can bookmark disliked comments?

Comment (dcDalin) [reply authors/apps/articles/serializers.py:78]: Yes, you can @ridgekimani, bookmarking acts as a 'temporary' storage for articles you may want to re-read in future.

**Pull request merged**